### PR TITLE
ci: Add semantic.yml for validation of commit msg

### PR DESCRIPTION
**Pull request by rsdmike**
Prepares for configuration of bot to address issue #2671

Signed-off-by: Mike <michael.johanson@intel.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit subject follows the [Conventional Commits spec](https://github.com/zeke/semantic-pull-requests)
- [x] The commit message follows the [EdgeX Contributor Guide](https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `make test` has completed successfully

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2671 

## What is the new behavior?
Will enforce commit message and PR title to follow conventional commit standard - customized to meet our scopes.
https://www.conventionalcommits.org/en/v1.0.0/#specification

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
No

## Are there any specific instructions or things that should be known prior to reviewing?
No
## Other information

Usage of this semantic.yml will be leveraged by https://github.com/apps/semantic-pull-requests once installed. A ticket will need to be created with LF in order to begin enforcement.

### CODE REVIEW

File: .github/semantic.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - fix
+    - docs
+    - style
+    - refactor
+    - perf
+    - test
+    - build
+    - ci
+    - revert
+scopes:
```
Comment (lenny-intel) [inline .github/semantic.yml:15]: How about also having `Common` or `Internal` for the common code in https://github.com/edgexfoundry/edgex-go/tree/master/internal/pkg

File: .github/semantic.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - core-command
+    - command
+    - cmd
+    - core-metadata
+    - metadata
+    - meta
+    - snap
+    - docker
+    - security
+    - scheduler
```
Comment (lenny-intel) [inline .github/semantic.yml:27]: Missing `notifications`

**Review by jpwhitemn — COMMENTED** [review]
other than @lenny-intel call on notifications.  LGTM

**Comment by jpwhitemn** [comment]
@rsdmike - given my PR #2760 , I think this PR is OBE.  Can we close?

**Comment by jpwhitemn** [comment]
already taken care of with #2760

**Pull request closed without merging**